projects/hangman/hangman_functions.py

```diff
@@ -0,0 +1,219 @@
+"""
+    Script for making functions for hangman game
+    :file: hangman_functions.py
+    :platform: Windows 10
+    :synopsis:
+        This script contains all the necessary functions
+        and variables that make up the whole game.
+
+    :author:
+        Muhammad Faheem-ur-Rehman
+        email: faheemlasani1034@gmail.com
+"""
+
+#importing required libraries
+from random import shuffle
+import string
+
+
+def check_word(user_letter, guess_list, memory_list, used_letters):
+    """
+       Function for getting combination of lists and judge that either user has put right alphabet or not
+
+       :param string user_letter: This string contains the immediate alphabet that user has put into
+       :param list guess_list: This list contains the alphabets of word that user is trying to guess
+       :param list memory_list: This list first contains only asterisks. Later on it gets updated
+       :param list used_letters : This list contains all alphabets that are already used by user
+
+       :returns:  True or False depending on that the guessed letter is right or wrong respectively
+       :return type: Boolean
+    """
+    check = False
+
+    for letters in user_letter:
+        if letters.islower():
+            user_letter = user_letter.upper()
+        else:
+            pass
+
+    print('                                                    ', end=' ')
+
+    for i in range(len(guess_list)):
+        if user_letter == guess_list[i]:
+            memory_list[i] = user_letter
+            check = 'True'
+        else:
+            pass
+
+    for letters in memory_list:
+        print(letters, end=' ')
+
+    if check:
+        print('             YESS; YOU GOT RIGHT', end=' ')
+    else:
+        print('             OOPPSSS; YOU MISSED', end=' ')
+
+    print('                 USED LETTERS:', end=' ')
+    for letters in used_letters:
+        if letters.isupper():
+            print(letters, end=' ')
+
+    print('\n')
+
+    return check
+
+
+def greeting_function():
+    """
+       Function for printing a greeting message and instructions to play the game
+
+       :param : none
+
+       :returns:  None
+       :return type: None
+    """
+    flag = True
+    print('\n')
+    print('                                                       *** WELCOME TO HANGMAN GAME ***')
+    print('\n')
+    print('                                              <><><> LETS UNDERSTAND HOW TO PLAY THE GAME <><><>')
+    print('\n')
```

Comment:
i would suggest to use pythonic way of giving spaces 
you can check this topic here https://pyformat.info/

![image](https://user-images.githubusercontent.com/12052941/104838106-1afe4e80-58b9-11eb-89db-ec48a414344b.png)

Reply:
Done